File: employee-management-system-ui-react/src/services/EmployeeService.js
import axios from "axios";

const BASE_API_URL = "http://localhost:8080/employees"

class EmployeeService{
    addEmployee(employee){
        return axios.post(BASE_API_URL, employee);
    }

    getAllEmployees(){
        return axios.get(BASE_API_URL);
    }

    deleteEmployee(id){
        return axios.delete(BASE_API_URL + "/" + id);
    }

    getEmployeeById(id){
        return axios.get(BASE_API_URL + "/" + id);
    }
    
    updateEmployee(id, employee){
        return axios.put(BASE_API_URL + "/" + id, employee);
    }
}

export default new EmployeeService();

File: Employee-Management-System-backend/src/main/java/com/spring/employeemanagementsystem/services/EmployeeService.java
package com.spring.employeemanagementsystem.services;

import com.spring.employeemanagementsystem.entities.Employee;
import org.springframework.stereotype.Service;

import java.util.List;


public interface EmployeeService {

    void addEmployee(Employee employee);

    List<Employee> getAllEmployees();

    boolean deleteEmployee(Long id);

    Employee getEmployeeById(Long id);

    boolean updateEmployee(Long id, Employee employee);
}
